domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsGreaterThanRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,447 @@
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_negativeFractionalInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_positiveMixedInput_exactSameValue_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_positiveMixedInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_positiveMixedInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_negativeMixedInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_positiveFractionalInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_positiveMixedInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveMixedAnswer_negativeFractionalInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_negativeMixedInput_answerGreaterThan_verifyAnswerGreater() {
```

Comment:
Move this up, along with other `testPositiveFractionalAnswer` cases.

Reply:
done